contrib/rabbitmq/pkg/reconciler/rabbitmqsource.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rabbitmq
+
+import (
+	"context"
+	"fmt"
+	"github.com/knative/eventing-contrib/contrib/rabbitmq/pkg/apis/sources/v1alpha1"
+	"github.com/knative/eventing-contrib/pkg/controller/sdk"
+	"github.com/knative/eventing-contrib/pkg/controller/sinks"
+	"github.com/knative/eventing-contrib/contrib/rabbitmq/pkg/reconciler/resources"
+	"knative.dev/pkg/logging"
+	"go.uber.org/zap"
+	"k8s.io/api/apps/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"log"
+	"os"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const (
+	controllerAgentName = "rabbitmq-source-controller"
+	raImageEnvVar       = "RABBITMQ_RA_IMAGE"
+	finalizerName       = controllerAgentName
+)
+
+func Add(mgr manager.Manager, logger *zap.SugaredLogger) error {
+	raImage, defined := os.LookupEnv(raImageEnvVar)
+	if !defined {
+		return fmt.Errorf("required environment variable '%s' not defined", raImageEnvVar)
+	}
+
+	log.Println("Adding the Rabbitmq Source controller.")
+	p := &sdk.Provider{
+		AgentName:  controllerAgentName,
+		Parent:     &v1alpha1.RabbitmqSource{},
+		Owns:       []runtime.Object{&v1.Deployment{}},
+		Reconciler: &reconciler{
+			scheme:              mgr.GetScheme(),
+			receiveAdapterImage: raImage,
+		},
+	}
+
+	return p.Add(mgr, logger)
+}
+
+type reconciler struct {
+	client              client.Client
+	scheme              *runtime.Scheme
+	receiveAdapterImage string
+}
+
+func (r *reconciler) InjectClient(c client.Client) error {
+	r.client = c
+	return nil
+}
+
+func (r *reconciler) Reconcile(ctx context.Context, object runtime.Object) error {
+	logger := logging.FromContext(ctx).Desugar()
```

Comment:
:smiley: 

If you want, you can use `knative/eventing/pkg/logging` (instead of knative.dev/pkg/logging`), which is already desugared.